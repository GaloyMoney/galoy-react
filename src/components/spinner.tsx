type Props = { size?: SpinnerSize }

export const Spinner = ({ size = "small" }: Props) => {
  if (size === "big") {
    return (
      <div className="spinner">
        <i aria-hidden className={`fas fa-spinner fa-spin fa-5x`} />
      </div>
    )
  }
  return <i aria-hidden className={`spinner fas fa-spinner fa-spin fa-lg`} />
}
