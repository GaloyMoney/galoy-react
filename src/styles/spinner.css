.spinner {
  animation: rotate 2s linear infinite;
  z-index: 2;

  & .path {
    stroke: #000000;
    stroke-linecap: round;
    animation: dash 1.5s ease-in-out infinite;
  }
}
