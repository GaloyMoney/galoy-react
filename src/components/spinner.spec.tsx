import { render } from "@testing-library/react"

import { Spinner } from "./spinner"

describe("Spinner", () => {
  it("renders and matches snapshot", () => {
    const { asFragment } = render(<Spinner />)
    expect(asFragment()).toMatchSnapshot()
  })
})
