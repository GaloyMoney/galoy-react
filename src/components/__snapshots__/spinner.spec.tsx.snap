// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders and matches snapshot 1`] = `
<DocumentFragment>
  <i
    aria-hidden="true"
    class="spinner fas fa-spinner fa-spin fa-lg"
  />
</DocumentFragment>
`;
